File: clean_code_study/230222_01_fp_oop_dependency_inversion/java/non_dip_way/Main.java
class Main {
    public static void main(String[] args) {
        App.start();
    }
}

class App {
    public static void start(){
        HttpClient.getUser("bob@clean.com");
        User user = new User("Bob", "bob@clean.com");
        HttpClient.createUser(user);
    }
}

class HttpClient {
    public static void getUser(String email){
        System.out.println("Getting user... " + email);
    };

    public static void createUser(User user){
        System.out.println("Creating user... " + user.name);
    };
}

class User{
    public String name;
    public String email;

    public User(String name, String email){
        this.name = name;
        this.email = email;
    }
}

File: clean_code_study/230222_01_fp_oop_dependency_inversion/go/non_dip_way/main.go
package main

import "fmt"

func main(){
	// calling phase
	app()
}

// declaration phase
func app(){
	httpclient := HttpClient{}
	httpclient.getUser("bob@clean.com")
	user := User{
		name: "Bob",
		email: "bob@clean.com",
	} 
	httpclient.createUser(user)
}

type HttpClient struct {
}

func (c HttpClient) getUser(email string){
	fmt.Println("Getting user...", email)
}

func (c HttpClient) createUser(user User){
	fmt.Println("Creating user...", user.name)
}

type User struct {
	name string
	email string
}

